.github/workflows/python-versions.yml

```diff
@@ -0,0 +1,51 @@
+name: Test Python Versions
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - '**'
+
+jobs:
+  python_tests:
+
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read 
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7.12", "3.10.0"]
```

Comment:
Small thing worth checking: do things still work if we only specify 3.7 / 3.10 here?

If so, that'd be slightly preferable since I would imagine in that case, `action/setup-python@v4` will install the newest patch version of the specified minor version, and I think we'd most likely want to do that over fixing a patch version.

Reply:
Tested & confirmed - shortening to  `python-version: ["3.7", "3.10"]` sets up v3.7.13 & v3.10.6 - will update 👍🏼